### PR TITLE
chore(project): add initial structure

### PR DESCRIPTION
**Pull request by joshblack**

### CODE REVIEW

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Communication through any of Carbon's channels (GitHub, emails, Twitter, Slack) must be constructive and never resort to personal attacks, trolling, public or private harassment, insults, or other unprofessional conduct.
+
+We promise to extend courtesy and respect to everyone involved in this project, regardless of gender, gender identity, sexual orientation, disability, age, race, ethnicity, religion, or level of experience. We expect anyone contributing to Carbon to do the same.
+
+If any member of the community violates this Code of Conduct, the maintainers of Carbon may take action, including removing issues, comments, and PRs or blocking accounts, as deemed appropriate.
+
+If you are subjected to or witness unacceptable behavior, or have any other concerns, please communicate with us directly via carbon@us.ibm.com.
+
+### Slack
```
Comment (photodow) [inline .github/CODE_OF_CONDUCT.md:21]: Does carbon have public slack channels?

Comment (joshblack) [reply .github/CODE_OF_CONDUCT.md:21]: @photodow unfortunately not 😞

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+   See [GitHub docs](https://help.github.com/articles/fork-a-repo/) for more details on forking a repository.
+
+3. Once cloned, you will see `origin` as your default remote, pointing to your personal forked repository. Add a remote named `upstream` pointing to the main `carbon-addons-dotcom`:
+
+   ```
+   $ git remote add upstream git@github.com:IBM/carbon-addons-dotcom.git
+   $ git remote -v
+   ```
+
+4. Switch to our version of Node. If you are using (nvm [Node Version Manager)](https://github.com/creationix/nvm), you can run `nvm use` to quicky switch Node versions. If you are aren't using nvm, you can still check the `.nvmrc` dotfile in the root of this project to find the major stable version of Node we are using, and then you can switch manually.
```
Comment (photodow) [inline .github/CONTRIBUTING.md]: Should the parenthesis be outside of the URL markup?

`(nvm [Node Version Manager)](...) --> (nvm [Node Version Manager](...)) `

Comment (photodow) [reply .github/CONTRIBUTING.md]: Based on this paragraph, it looks like we'll need to add a `.nvmrc` file.

**Review by kennylam — APPROVED** [review]
Looks good thanks @joshblack

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+7. Once ready for feedback from other contributors and maintainers, **push your commits to your fork** (be sure to run `yarn ci-check` before pushing, to make sure your code passes linting and unit tests):
+
+   ```
+   $ git push origin { YOUR_BRANCH_NAME }
+   ```
+
+8. In Github, navigate to [IBM/carbon-addons-dotcom](https://github.com/IBM/carbon-addons-dotcom) and click the button that reads "Compare & pull request".
+
+9. Write a title and description, the click "Create pull request".
```
Comment (photodow) [inline .github/CONTRIBUTING.md]: Is that meant to be `then click` instead of `the click`?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+   See [how to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request) for more details on writing good PRs.
+
+10. Stay up to date with the activity in your pull request. Maintainers will be reviewing your work and making comments, asking questions and suggesting changes to be made before they merge your code. When you need to make a change, add, commit and push to your branch normally.
+
+    Once all revisions to your pull request are complete, a maintainer will squash and merge your commits for you.
+
+For all details related to development, make sure to check out our [development
+guide](/docs/developing.md)!
```
Comment (photodow) [inline .github/CONTRIBUTING.md]: Looks like we're going to need to add this file.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+> Is this a feature request (new component, new icon), a bug, or a general issue?
+
+> Is this issue related to a specific component?
+
+> What did you expect to happen? What happened instead? What would you like to see changed?
+
+> What browser are you working in?
+
+> What version of the Carbon Design System are you using?
```
Comment (photodow) [inline .github/ISSUE_TEMPLATE/bug_report.md:24]: @joshblack should this be carbon design system, carbon-addons-dotcom, or just remove that reference altogether?

File: .github/ISSUE_TEMPLATE/question.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+---
+name: Question 🤔
+about: Usage question or discussion about the Carbon Design System.
```
Comment (photodow) [inline .github/ISSUE_TEMPLATE/question.md:3]: @joshblack Again, curious if this is meant to be at the Carbon Design System level, or if we should get more specific.

**Review by photodow — APPROVED** [review]
I had one or two questions, feel free to answer when you get time. This looks amazing otherwise. Thanks for the leg up!

**Pull request merged**